pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -0,0 +1,473 @@
+	return taskRunCondSet.Manage(trs).GetCondition(t)
+}
+
+// InitializeConditions will set all conditions in taskRunCondSet to unknown for the TaskRun
+// and set the started time to the current time
+func (trs *TaskRunStatus) InitializeConditions() {
+	started := false
+	if trs.StartTime.IsZero() {
+		trs.StartTime = &metav1.Time{Time: time.Now()}
+		started = true
+	}
+	conditionManager := taskRunCondSet.Manage(trs)
+	conditionManager.InitializeConditions()
+	// Ensure the started reason is set for the "Succeeded" condition
+	if started {
+		initialCondition := conditionManager.GetCondition(apis.ConditionSucceeded)
+		initialCondition.Reason = TaskRunReasonStarted.String()
+		conditionManager.SetCondition(*initialCondition)
+	}
+}
+
+// SetCondition sets the condition, unsetting previous conditions with the same
+// type as necessary.
+func (trs *TaskRunStatus) SetCondition(newCond *apis.Condition) {
+	if newCond != nil {
+		taskRunCondSet.Manage(trs).SetCondition(*newCond)
+	}
+}
+
+// StepState reports the results of running a step in a Task.
+type StepState struct {
+	corev1.ContainerState `json:",inline"`
+	Name                  string `json:"name,omitempty"`
+	ContainerName         string `json:"container,omitempty"`
+	ImageID               string `json:"imageID,omitempty"`
+}
+
+// SidecarState reports the results of running a sidecar in a Task.
+type SidecarState struct {
+	corev1.ContainerState `json:",inline"`
+	Name                  string `json:"name,omitempty"`
+	ContainerName         string `json:"container,omitempty"`
+	ImageID               string `json:"imageID,omitempty"`
+}
+
+// CloudEventDelivery is the target of a cloud event along with the state of
+// delivery.
+type CloudEventDelivery struct {
+	// Target points to an addressable
+	Target string                  `json:"target,omitempty"`
+	Status CloudEventDeliveryState `json:"status,omitempty"`
+}
+
+// CloudEventCondition is a string that represents the condition of the event.
+type CloudEventCondition string
+
+const (
+	// CloudEventConditionUnknown means that the condition for the event to be
+	// triggered was not met yet, or we don't know the state yet.
+	CloudEventConditionUnknown CloudEventCondition = "Unknown"
+	// CloudEventConditionSent means that the event was sent successfully
+	CloudEventConditionSent CloudEventCondition = "Sent"
+	// CloudEventConditionFailed means that there was one or more attempts to
+	// send the event, and none was successful so far.
+	CloudEventConditionFailed CloudEventCondition = "Failed"
+)
+
+// CloudEventDeliveryState reports the state of a cloud event to be sent.
+type CloudEventDeliveryState struct {
+	// Current status
+	Condition CloudEventCondition `json:"condition,omitempty"`
+	// SentAt is the time at which the last attempt to send the event was made
+	// +optional
+	SentAt *metav1.Time `json:"sentAt,omitempty"`
+	// Error is the text of error (if any)
+	Error string `json:"message"`
```

Comment:
it looks like there's an API rule violation here, and I'm not sure whether it makes sense to rename the API field here or just the Go struct. Is this message expected to be present under non-error conditions or not? @afrittoli wdyt?

Reply:
`CloudEventDeliveryState` is only used for the `CloudEventPipelineResource` so given the fact that we are not having `PipelineResources` in `v1` both the API and structs should go away in `v1`.

Reply:
Thanks @afrittoli. Removed both from v1.

Reply:
Could you please update the commit message and PR description to reflect that these structs have been removed, that you renamed the StepState.ContainerName and SidecarState.ContainerName Go fields (but not the api fields), and that the "taskResults" api field was renamed to "results"

Reply:
Updated. Thanks Lee @lbernick  for the summary.